test/utils/validateMemos.test.ts

```diff
@@ -0,0 +1,135 @@
+        denominations: [],
+        displayName: 'Memo Test Coin',
+        pluginId: 'memoTest',
+        transactionExplorer: '',
+        walletType: 'wallet:memoTest',
+        ...extraInfo
+      }
+    )
+  }
+
+  it('rejects multiple memos', function () {
+    expect(() =>
+      callValidateMemos(
+        [
+          { type: 'hex', value: '' },
+          { type: 'hex', value: '' }
+        ],
+        { memoOptions: [{ type: 'hex' }] }
+      )
+    ).to.throw('Memo Test Coin only supports one memo')
+  })
+
+  it('accepts multiple memos', function () {
+    callValidateMemos(
+      [
+        { type: 'text', value: '' },
+        { type: 'hex', value: '' }
+      ],
+      {
+        memoOptions: [{ type: 'hex' }, { type: 'text' }],
+        multipleMemos: true
+      }
+    )
+  })
+
+  it('validates hex memos', function () {
+    const extraInfo: Partial<EdgeCurrencyInfo> = {
+      memoOptions: [
+        {
+          type: 'hex',
+          memoName: 'data',
+          minBytes: 2,
+          maxBytes: 4
+        }
+      ]
+    }
+
+    // Good:
+    callValidateMemos([{ type: 'hex', value: '12345678' }], extraInfo)
+
+    // Bad:
+    expect(() =>
+      callValidateMemos([{ type: 'hex', value: '0xff' }], extraInfo)
+    ).throws('Memo Test Coin data: is not valid hexadecimal')
+    expect(() =>
+      callValidateMemos([{ type: 'hex', value: 'ff' }], extraInfo)
+    ).throws('Memo Test Coin data: cannot be shorter than 2 bytes')
+    expect(() =>
+      callValidateMemos([{ type: 'hex', value: 'ff12345678' }], extraInfo)
+    ).throws('Memo Test Coin data: cannot be longer than 4 bytes')
+    expect(() =>
+      callValidateMemos([{ type: 'text', value: 'hello' }], extraInfo)
+    ).throws('Memo Test Coin data: cannot be type text')
+  })
+
+  it('validates number memos', function () {
+    const extraInfo: Partial<EdgeCurrencyInfo> = {
+      memoOptions: [
+        {
+          type: 'number',
+          memoName: 'tag',
+          maxValue: '256'
+        }
+      ]
+    }
+
+    // Good:
+    callValidateMemos([{ type: 'number', value: '255' }], extraInfo)
+    callValidateMemos([{ type: 'number', value: '256' }], extraInfo)
+
+    // Bad:
+    expect(() =>
+      callValidateMemos([{ type: 'number', value: '257' }], extraInfo)
+    ).throws('Memo Test Coin tag: cannot be greater than 256')
+    expect(() =>
+      callValidateMemos([{ type: 'number', value: '1b' }], extraInfo)
+    ).throws('Memo Test Coin tag: is not a valid number')
+    expect(() =>
+      callValidateMemos([{ type: 'hex', value: '12345678' }], extraInfo)
+    ).throws('Memo Test Coin tag: cannot be type hex')
+  })
+
+  it('validates string memos', function () {
+    const extraInfo: Partial<EdgeCurrencyInfo> = {
+      memoOptions: [
+        {
+          type: 'text',
+          memoName: 'message',
+          maxLength: 6
+        }
+      ]
+    }
+
+    // Good:
+    callValidateMemos([{ type: 'text', value: 'hello' }], extraInfo)
```

Comment:
I'd also add a case for a `hex` here. That way we can catch when implementation confuses strings with hex. I'm thinking `"0x123"` and `cbdc` as a couple of cases that are valid hex and also valid strings